### PR TITLE
[15.0][IMP] product_assortment: Add security group to only show assortments to managers

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT51064

ping @CarlosRoca13 @carlos-lopez-tecnativa

### CODE REVIEW

File: product_assortment/security/product_assortment_security.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo noupdate="1">
+    <record id="group_product_assortment_manager" model="res.groups">
```
Comment (carlos-lopez-tecnativa) [inline product_assortment/security/product_assortment_security.xml:3]: If the intention is to restrict the `Product Assortment` menu to managers, I think it’s necessary to add the new group to the button in the Partner form view as well. Or did I misunderstand the PR's purpose? What do you think?
image

**Comment by sergio-teruel** [comment]
Hi @carlos-lopez-tecnativa I agree...

Changes done!

**Review by carlos-lopez-tecnativa — APPROVED** [review]
LGTM thanks

**Review by CarlosRoca13 — APPROVED** [review]

File: product_assortment/security/product_assortment_security.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        <field name="name">Product Assortment Manager</field>
+        <field name="comment">Access to all product assortment</field>
+    </record>
+    <record id="base.ir_filters_employee_rule" model="ir.rule">
+        <field
+            name="domain_force"
+        >[('user_id','in',[False,user.id]), ('is_assortment', '=', False)]</field>
```
Comment (pedrobaeza) [inline product_assortment/security/product_assortment_security.xml]: @sergio-teruel I think this is not very resilient, as other modules modifying the same rule will make this fail. What about overriding the `ir.rule` `_compute_domain` to inject the extra domain for this one? Example: https://github.com/OCA/sale-workflow/blob/16.0/sales_team_security/models/ir_rule.py#L23

**Comment by sergio-teruel** [comment]
Changes Done!!

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge minor

**Comment by pedrobaeza** [comment]
Please fw-port it to upper versions.

**Pull request merged**